src/Fhir.Anonymizer.Tool/FhirResourceDataProcessor.cs

```diff
@@ -74,33 +77,34 @@ public void AnonymizeBulkDataFolder(string inputFolder, string outputFolder, boo
                     Directory.CreateDirectory(resourceOutputFolder);
                 }
 
-                var processedCount = 0;
-                var processedErrorCount = 0;
+                int completedCount = 0;
+                int failedCount = 0;
+                int consumeCompletedCount = 0;
                 using (FileStream inputStream = new FileStream(bulkResourceFileName, FileMode.Open))
                 using (FileStream outputStream = new FileStream(bulkResourceOutputFileName, FileMode.Create))
                 {
-                    using StreamReader reader = new StreamReader(inputStream);
-                    using StreamWriter writer = new StreamWriter(outputStream);
+                    using FhirStreamReader reader = new FhirStreamReader(inputStream);
```

Comment:
We need to think about forward compatibility with reader for XML input. Looks like it's fine.

Reply:
Yes, the only change we need to apply xml to xml is to implement a new anonymizer function without partition executor change.